--- lz4.h.orig	2013-03-27 13:04:54.792931974 +0100
+++ lz4.h	2013-03-28 08:27:30.997919652 +0100
@@ -37,21 +37,41 @@
 extern "C" {
 #endif
 
+/* optional conf.h file if build with -DFASTLZ_INCLUDE_CONF_H */
+#ifdef LZ4_INCLUDE_CONF_H
+#include "conf.h"
+#endif
+#ifndef ZFASTEXTERN
+#ifdef _WIN32
+#ifdef FASTLZ_DLL
+#define ZFASTEXTERN __declspec(dllexport)
+#else
+#define ZFASTEXTERN __declspec(dllimport)
+#endif
+#else
+#define ZFASTEXTERN extern
+#endif
+#define ZFASTINLINE
+#endif
 
-//**************************************
-// Compiler Options
-//**************************************
-#if defined(_MSC_VER) && !defined(__cplusplus)   // Visual Studio
-#  define inline __inline           // Visual is not C99, but supports some kind of inline
+/***************************************
+  Compiler Options
+ **************************************/
+#ifndef LZ4_INCLUDE_CONF_H
+  /* Visual Studio */
+#if ((defined(_MSC_VER)) && (!defined(__cplusplus)))
+#  undef ZFASTINLINE
+#  define ZFASTINLINE __inline           // Visual is not C99, but supports some kind of inline
+#endif
 #endif
 
 
-//****************************
-// Simple Functions
-//****************************
+/****************************
+  Simple Functions
+ ****************************/
 
-int LZ4_compress   (const char* source, char* dest, int isize);
-int LZ4_uncompress (const char* source, char* dest, int osize);
+ZFASTEXTERN int LZ4_compress   (const char* source, char* dest, int isize);
+ZFASTEXTERN int LZ4_uncompress (const char* source, char* dest, int osize);
 
 /*
 LZ4_compress() :
@@ -74,11 +94,11 @@
 */
 
 
-//****************************
-// Advanced Functions
-//****************************
+/****************************
+  Advanced Functions
+ ****************************/
 
-static inline int LZ4_compressBound(int isize)   { return ((isize) + ((isize)/255) + 16); }
+static ZFASTINLINE int LZ4_compressBound(int isize)   { return ((isize) + ((isize)/255) + 16); }
 #define           LZ4_COMPRESSBOUND(    isize)            ((isize) + ((isize)/255) + 16)
 
 /*
@@ -94,7 +114,7 @@
 */
 
 
-int LZ4_compress_limitedOutput   (const char* source, char* dest, int isize, int maxOutputSize);
+ZFASTEXTERN int LZ4_compress_limitedOutput   (const char* source, char* dest, int isize, int maxOutputSize);
 
 /*
 LZ4_compress_limitedOutput() :
@@ -109,7 +129,7 @@
 */
 
 
-int LZ4_uncompress_unknownOutputSize (const char* source, char* dest, int isize, int maxOutputSize);
+ZFASTEXTERN int LZ4_uncompress_unknownOutputSize (const char* source, char* dest, int isize, int maxOutputSize);
 
 /*
 LZ4_uncompress_unknownOutputSize() :
