--- lz4.h.orig	2013-03-27 11:54:15.408398508 +0100
+++ lz4.h	2013-03-27 11:59:06.235173196 +0100
@@ -37,21 +37,39 @@
 extern "C" {
 #endif
 
+/* optional conf.h file if build with -DFASTLZ_INCLUDE_CONF_H */
+#ifdef LZ4_INCLUDE_CONF_H
+#include "conf.h"
+#endif
+#ifndef ZFASTEXTERN
+#ifdef _WIN32
+#ifdef FASTLZ_DLL
+#define ZFASTEXTERN __declspec(dllexport)
+#else
+#define ZFASTEXTERN __declspec(dllimport)
+#endif
+#else
+#define ZFASTEXTERN extern
+#endif
+#define ZFASTINLINE
+#endif
 
 //**************************************
 // Compiler Options
 //**************************************
+#ifndef LZ4_INCLUDE_CONF_H
 #if defined(_MSC_VER) && !defined(__cplusplus)   // Visual Studio
 #  define inline __inline           // Visual is not C99, but supports some kind of inline
 #endif
+#endif
 
 
 //****************************
 // Simple Functions
 //****************************
 
-int LZ4_compress   (const char* source, char* dest, int isize);
-int LZ4_uncompress (const char* source, char* dest, int osize);
+ZFASTEXTERN int LZ4_compress   (const char* source, char* dest, int isize);
+ZFASTEXTERN int LZ4_uncompress (const char* source, char* dest, int osize);
 
 /*
 LZ4_compress() :
@@ -94,7 +112,7 @@
 */
 
 
-int LZ4_compress_limitedOutput   (const char* source, char* dest, int isize, int maxOutputSize);
+ZFASTEXTERN int LZ4_compress_limitedOutput   (const char* source, char* dest, int isize, int maxOutputSize);
 
 /*
 LZ4_compress_limitedOutput() :
@@ -109,7 +127,7 @@
 */
 
 
-int LZ4_uncompress_unknownOutputSize (const char* source, char* dest, int isize, int maxOutputSize);
+ZFASTEXTERN int LZ4_uncompress_unknownOutputSize (const char* source, char* dest, int isize, int maxOutputSize);
 
 /*
 LZ4_uncompress_unknownOutputSize() :
